internal/controller/backup_controller.go

```diff
@@ -287,7 +287,10 @@ func (r *BackupReconciler) isValidBackupRunning(
 		return false, fmt.Errorf("unknown.spec.target received: %s", backup.Spec.Target)
 	}
 
-	containerIsNotRestarted := backup.Status.InstanceID.ContainerID == pod.Status.ContainerStatuses[0].ContainerID
```

Comment:
This could be simplified. At this line, we're sure that `backup.Status.InstanceID != nil` otherwise, we returned a while ago.

If we find that this Pod has no container statuses, then it is new.
If it is new, it is not the same Pod that was taking the backup.
Therefore, this backup has been restarted.